lib/mix/test/mix/tasks/test_test.exs

```diff
@@ -624,6 +624,33 @@ defmodule Mix.Tasks.TestTest do
     end
   end
 
+  describe "test_load_filters and test_warn_filters" do
+    test "warns for files that are not loaded and don't match test_warn_filters" do
+      in_fixture("test_warn", fn ->
```

Comment:
Instead of adding a new fixture, prefer to generate the files inside the test. Having the fixtures generated inside the test is easier to maintain (and we generally use too many fixtures in this repository already). Both test_stale and test_failed should not be fixtures either, the fixture makes sense only if they are shared across files, but I guess that's a separate conversation.